### ISSUE TITLE
Ambiguous message: "You attempted to join using a room ID without providing a list of servers to join through."

### ISSUE THREAD

**Issue by rltas** [vector-im/element-web#25627]
### Steps to reproduce

1. Create a room.
2. Leave the room.
3. Try to "Join the discussion" again from "Historical".

#### What did you expect?

A message telling me that the room doesn't exist / can't be found on the server.

#### What happened instead?

A message telling me something that confuses me:

> **Failed to join**
> You attempted to join using a room ID without providing a list of servers to join through. Room IDs are internal identifiers and cannot be used to join a room without additional information.
> If you know a room address, try joining through that instead.

#### Additional information

In the browser console I see

> `POST https://matrix-client.matrix.org/_matrix/client/r0/join/!NVfMslxkPhhjsSXnGH%3Amatrix.org 404`

Not sure if it's always been like that or only recently because maybe [forget_rooms_on_leave](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#forget_rooms_on_leave) has been enabled for matrix.org?

### URL for webapp

app.element.io

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by rltas** [comment]
Yes. One of our users stumbled upon this and I reproduced it. The problem isn't that the room can't be joined again, but that the message is ambiguous.

I just found https://github.com/vector-im/element-meta/issues/419 and https://github.com/matrix-org/matrix-react-sdk/pull/6048, but that was about permalinks from what I see. The message shouldn't appear in the situation described above imo.

**Comment by weeman1337** [comment]
Can reproduce as well.

**Comment by digit4lsh4d0w** [comment]
I also ran into this error. I'm in the room `#homeowners:matrix.org`, where the ID of the support room is given in the subject: `#synapse:matrix.org`.

Both Element Web and Element Desktop display the same error when trying to join a room using this identifier. I tried to find and join the room through the compass icon.

It turned out to join only with the help of a manually created URL and matrix.to . Here is an example: `https://matrix.to/#/#synapse:matrix.org?via={{SOME_HOME_SERVER}}`.

All attempts were made through our own home server and our own Element Web.

Application versions:

- Element Web: Docker 1.11.110
- Element Desktop: ArchLinux 1.11.110-1
- Synapse: Docker 1.138.0
